### ISSUE TITLE
Prereq requisite does not work in orchestrations

### ISSUE THREAD

**Issue by amendlik** [saltstack/salt#16969]
The 'prereq' requisite does not work with the state.orchestrate runner. It does reorder the steps (as if require_in was specified) but does not cause the prerequisite state to execute conditionally.

Example:
```
server.down:
  salt.state:
    - tgt: 'web1'
    - prereq:
      - salt: server.deploy

server.deploy:
  salt.state:
    - tgt: 'web1'
```

I would expect the server.down state to only execute if the server.deploy state was going to change something. Instead, server.down runs every time, regardless.

**Comment by jfindlay** [comment]
@basepi, any progress on this?

**Comment by cachedout** [comment]
Fixed by #25587
